python/pyspark/sql/types.py

```diff
@@ -1236,6 +1230,7 @@ class Row(tuple):
 
         """ Row in DataFrame """
         __datatype = dataType
```

Comment:
can you add inline comment for all the private fields here on what they do? 

this includes __datatype, __converters__, __fields__, __slots__